common/config/bootstrap.go

```diff
@@ -0,0 +1,149 @@
+package config
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+
+	"github.com/Layr-Labs/eigenda/common"
+	"github.com/Layr-Labs/eigenda/common/pprof"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/urfave/cli/v2"
+)
+
+var (
+	pprofFlag = &cli.BoolFlag{
+		Name:    "pprof",
+		Aliases: []string{"p"},
+		Usage:   "If set, starts a pprof server.",
+	}
+	pprofPortFlag = &cli.IntFlag{
+		Name:    "pprof-port",
+		Aliases: []string{"o"},
+		Usage:   "Port for the pprof server.",
+		Value:   6060,
+	}
+	debugFlag = &cli.BoolFlag{
+		Name:    "debug",
+		Aliases: []string{"d"},
+		Usage:   "Enable debug mode. Program will pause for a debugger to attach.",
+	}
+	disableEnvVarsFlag = &cli.BoolFlag{
+		Name:    "disable-env-vars",
+		Aliases: []string{"e"},
+		Usage:   "Disable loading configuration from environment variables.",
+	}
+	overrideEnvPrefixFlag = &cli.StringFlag{
+		Name:    "env-prefix",
+		Aliases: []string{"r"},
+		Usage:   "If set, overrides the environment variable prefix used to load configuration from env vars.",
+	}
+	configFileFlag = &cli.StringSliceFlag{
+		Name:    "config",
+		Aliases: []string{"c"},
+		Usage:   "Path to a configuration file. Can be specified multiple times to load multiple files.",
+	}
+)
+
+// Reads command line arguments, loads configuration from files and environment variables as specified.
+func Bootstrap[T VerifiableConfig](
+	constructor func() T,
+	defaultEnvVarPrefix string,
+) (T, error) {
+
+	// We need a logger before we have a logger config. Once we parse config, we can initialize the real logger.
+	bootstrapLogger, err := common.NewLogger(common.DefaultConsoleLoggerConfig())
+	if err != nil {
+		var zero T
+		return zero, fmt.Errorf("failed to create bootstrap logger: %w", err)
+	}
+
+	action, cfgChan := buildHandler(bootstrapLogger, constructor, defaultEnvVarPrefix)
+
+	app := &cli.App{
+		Flags: []cli.Flag{
+			pprofFlag,
+			pprofPortFlag,
+			debugFlag,
+			disableEnvVarsFlag,
+			overrideEnvPrefixFlag,
+			configFileFlag,
+		},
+		Action: action,
+	}
+
+	err = app.Run(os.Args)
+	if err != nil {
+		var zero T
+		return zero, fmt.Errorf("error parsing command line arguments: %w", err)
+	}
+
+	// If the help flag was set, the action never runs and cfgChan is never written to.
```

Comment:
I don't see any actual implementation for `help`. Is that baked into urfave somehow?